### PR TITLE
opt: opt gl api use

### PR DESCRIPTION
**Pull request by singlecoder**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: packages/core/src/shader/ShaderProgram.ts [hunk 1]
```diff
@@ -262,10 +262,12 @@ export class ShaderProgram {
       return null;
     }
 
-    if (!gl.getProgramParameter(program, gl.LINK_STATUS)) {
-      Logger.error("Could not link WebGL program. \n" + gl.getProgramInfoLog(program));
-      gl.deleteProgram(program);
-      return null;
+    if (Logger.isEnabled) {
```
Comment (eyworldwide) [inline packages/core/src/shader/ShaderProgram.ts:265]: `Logger.isEnabled && !gl.getProgramParameter(program, gl.LINK_STATUS)` is better

Comment (singlecoder) [reply packages/core/src/shader/ShaderProgram.ts]: > `Logger.isEnabled && !gl.getProgramParameter(program, gl.LINK_STATUS)` is better

Done

**Review by GuoLei1990 — APPROVED** [review]
+1

**Review by GuoLei1990 — APPROVED** [review]
+1

**Pull request merged**